projects/resource-library/src/lib/components/library-filter/library-filter.component.ts

```diff
@@ -0,0 +1,167 @@
+import { Component, OnInit, Input, Output, EventEmitter, OnChanges, ViewEncapsulation } from '@angular/core';
+import * as _ from 'lodash-es';
+import { Subject } from 'rxjs';
+import { takeUntil, filter, take } from 'rxjs/operators';
+import { TreeService } from '../../services/tree/tree.service';
+import { EditorService } from '../../services/editor/editor.service';
+import { FrameworkService } from '../../services/framework/framework.service';
+import { EditorTelemetryService } from '../../services/telemetry/telemetry.service';
+import { ConfigService } from '../../services/config/config.service';
+import { HelperService } from '../../services/helper/helper.service';
+
+@Component({
+  selector: 'lib-library-filter',
+  templateUrl: './library-filter.component.html',
+  styleUrls: ['./library-filter.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LibraryFilterComponent implements OnInit, OnChanges {
+  @Input() sessionContext: any;
+  @Input() filterValues: any;
+  @Input() filterOpenStatus: boolean;
+  @Input() searchFormConfig: any;
+  @Input() frameworkId: any;
+  @Output() filterChangeEvent: EventEmitter<any> = new EventEmitter();
+  public filterConfig: any;
+  public isFilterShow = false;
+  public filterFields: any;
+  public telemetryPageId: string;
+  private onComponentDestroy$ = new Subject<any>();
+  public frameworkDetails: any = {};
+  public currentFilters: any;
+  public searchQuery: string;
+
+  constructor(private frameworkService: FrameworkService,
+              public editorService: EditorService,
+              public telemetryService: EditorTelemetryService,
+              public treeService: TreeService,
+              public configService: ConfigService,
+              private helperService: HelperService) { }
+
+  ngOnInit() {
+    this.filterFields = this.searchFormConfig;
+    /*const selectedNode = this.treeService.getActiveNode();
+    let contentTypes = _.flatten(
+      _.map(_.get(this.editorService.editorConfig.config, `hierarchy.level${selectedNode.getLevel() - 1}.children`), (val) => {
+      return val;
+    }));
+
+    if (_.isEmpty(contentTypes)) {
+      contentTypes = _.map(this.helperService.contentPrimaryCategories, 'name');
+    }*/
+    const contentTypes = _.map(this.helperService.contentPrimaryCategories, 'name');
+
+    this.currentFilters = {
+      primaryCategory: contentTypes,
+      board: [],
+      medium: [],
+      gradeLevel: [],
+      subject: [],
```

Comment:
Can we avoid hardcoding framework categories?

Reply:
fixed